contracts/smart-wallet/non-upgradeable/Account.sol

```diff
@@ -167,6 +172,24 @@ contract Account is
         return 0;
     }
 
+    /// @notice Registers a signer in the factory.
+    function _setupRole(bytes32 role, address account) internal virtual override {
+        super._setupRole(role, account);
+
+        if (role == SIGNER_ROLE) {
+            AccountFactory(factory).addSigner(account);
```

Comment:
This could revert if factory is address zero or not a contract

Reply:
I've updated the check: 

```solidity
if (role == SIGNER_ROLE && factory.code.length > 0) {
    AccountFactory(factory).addSigner(account);
}
```